### PR TITLE
Updated public API methods containing blockNumber

### LINKED ISSUE #1385: Update `blockNumber` parameter descriptions
For any public API method that contains the `blockNumber` parameter (for example, [`debug_getRawBlock`](https://besu.hyperledger.org/public-networks/reference/api#debug_getrawblock)), update the description to specify that it's a "hexadecimal integer." Also, include `finalized` and `safe` as possible values.

### PR DESCRIPTION
**Pull request by mjsmike62**
Updated public API methods containing the `blockNumber` parameter, specifying that they are `hexadecimal integer` and also added `finalized` and `safe` to the string tags listing. Also, added `finalized` and `safe` to other public API methods that were missing these tags.

Fixes #1385 

See https://besu-docs-1sz5lp094-hyperledger.vercel.app/development/public-networks/reference/api

- Made additional edits as per Alex's review comments below, "Missing a few updates". 
- Made additional edits as per Sally MacFarlane's review comments below, "blockNumber can be hex OR decimal "
- Made additional edits as per Sally MacFarlane's review comments below: changed 'decimal' to 'decimal integer'.

See:
https://besu-docs-lnswwc430-hyperledger.vercel.app/development/public-networks/reference/api
https://besu-docs-lnswwc430-hyperledger.vercel.app/development/private-networks/reference/api

### CODE REVIEW

**Review by alexandratran — COMMENTED** [review]
Missing a few updates:

- `admin_logsRemoveCache` `fromBlock` and `toBlock` missing "hexadecimal"
- `eth_feeHistory` `newestBlock` missing "hexadecimal"
- `debug_getRawReceipts` `blockNumber` missing both "hexadecimal" and the new string tags

**Comment by mjsmike62** [comment]
@alexandratran, the only place where I can find that `eth_feeHistory` is missing the **hexadecimal integer** description is in down-level versions of the docs.
eth_feeHistory
.

File: docs/public-networks/reference/api/index.md [hunk 1]
```diff
@@ -2282,7 +2282,7 @@ Returns base fee per gas and transaction effective priority fee per gas history
 
 - `blockCount`: _integer_ or _string_ - Number of blocks in the requested range. Between 1 and 1024 blocks can be requested in a single query. If blocks in the specified block range are not available, then only the fee history for available blocks is returned. Accepts hexadecimal or integer values.
 
-- `newestBlock`: _string_ - Integer representing the highest number block of the requested range or one of the string tags `latest`, `earliest`, or `pending`, as described in [block parameter](../../how-to/use-besu-api/json-rpc.md#block-parameter).
+- `newestBlock`: _string_ - Integer representing the highest number block of the requested range or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [block parameter](../../how-to/use-besu-api/json-rpc.md#block-parameter).
```
Comment (alexandratran) [inline docs/public-networks/reference/api/index.md:2285]: @mjsmike62 This is the `eth_feeHistory` parameter I'm referring to. Around line 2285 in this file.

**Review by macfarla — COMMENTED** [review]
blockNumber can be hex OR decimal - see
https://besu.hyperledger.org/development/public-networks/how-to/use-besu-api/json-rpc#block-parameter

so you could have "0x10" or "16" and either would work

File: docs/private-networks/reference/api/index.md [hunk 2]
```diff
@@ -67,7 +67,7 @@ Lists [signers for the specified block].
 
 #### Parameters
 
-`blockNumber`: _string_ - hexadecimal integer representing a block number or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [Block Parameter](../../../public-networks/how-to/use-besu-api/json-rpc.md#block-parameter)
+`blockNumber`: _string_ - hexadecimal or decimal representing a block number or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [Block Parameter](../../../public-networks/how-to/use-besu-api/json-rpc.md#block-parameter)
```
Comment (macfarla) [inline docs/private-networks/reference/api/index.md:70]:
```suggestion
`blockNumber`: _string_ - hexadecimal or decimal integer representing a block number, or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [Block Parameter](../../../public-networks/how-to/use-besu-api/json-rpc.md#block-parameter)
```

Comment (macfarla) [reply docs/private-networks/reference/api/index.md:70]: does that make it clearer?

**Review by macfarla — APPROVED** [review]
minor suggestion on wording

File: docs/private-networks/reference/api/index.md [hunk 3]
```diff
@@ -579,7 +579,7 @@ Lists the validators defined in the specified block.
 
 #### Parameters
 
-- `blockNumber`: _string_ - integer representing a block number or one of the string tags `latest`, `earliest`, or `pending`, as described in [Block Parameter](../../../public-networks/how-to/use-besu-api/json-rpc.md#block-parameter)
+- `blockNumber`: _string_ - hexadecimal or decimal integer representing a block number or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [Block Parameter](../../../public-networks/how-to/use-besu-api/json-rpc.md#block-parameter)
```
Comment (alexandratran) [inline docs/private-networks/reference/api/index.md:582]:
```suggestion
- `blockNumber`: _string_ - hexadecimal or decimal integer representing a block number or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [block parameter](../../../public-networks/how-to/use-besu-api/json-rpc.md#block-parameter)
```

File: docs/private-networks/reference/api/index.md [hunk 4]
```diff
@@ -991,7 +991,7 @@ For private contracts, `priv_call` is the same as [`eth_call`](../../../public-n
 
 - `call`: _object_ - [transaction call object](../../../public-networks/reference/api/objects.md#transaction-call-object)
 
-- `blockNumber`: _string_ - integer representing a block number or one of the string tags `latest`, `earliest`, or `pending`, as described in [Block Parameter](../../../public-networks/how-to/use-besu-api/json-rpc.md#block-parameter)
+- `blockNumber`: _string_ - hexadecimal or decimal integer representing a block number or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [Block Parameter](../../../public-networks/how-to/use-besu-api/json-rpc.md#block-parameter)
```
Comment (alexandratran) [inline docs/private-networks/reference/api/index.md:994]:
```suggestion
- `blockNumber`: _string_ - hexadecimal or decimal integer representing a block number or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [block parameter](../../../public-networks/how-to/use-besu-api/json-rpc.md#block-parameter)
```

File: docs/private-networks/reference/api/index.md [hunk 5]
```diff
@@ -1111,7 +1111,7 @@ Returns the state root of the specified privacy group at the specified block.
 
 - `privacyGroupId`: _string_ - 32-byte [privacy Group ID](../../concepts/privacy/privacy-groups.md)
 
-- `blockNumber`: _string_ - integer representing a block number or one of the string tags `latest`, `earliest`, or `pending`, as described in [Block Parameter](../../../public-networks/how-to/use-besu-api/json-rpc.md#block-parameter)
+- `blockNumber`: _string_ - hexadecimal or decimal integer representing a block number or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [Block Parameter](../../../public-networks/how-to/use-besu-api/json-rpc.md#block-parameter)
```
Comment (alexandratran) [inline docs/private-networks/reference/api/index.md:1114]:
```suggestion
- `blockNumber`: _string_ - hexadecimal or decimal integer representing a block number or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [block parameter](../../../public-networks/how-to/use-besu-api/json-rpc.md#block-parameter)
```

File: docs/private-networks/reference/api/index.md [hunk 6]
```diff
@@ -1288,7 +1288,7 @@ Returns the code of the private smart contract at the specified address. Compile
 
 - `address`: _string_ - 20-byte contract address
 
-- `blockNumber`: _string_ - integer representing a block number or one of the string tags `latest`, `earliest`, or `pending`, as described in [Block Parameter](../../../public-networks/how-to/use-besu-api/json-rpc.md#block-parameter)
+- `blockNumber`: _string_ - hexadecimal or decimal integer representing a block number or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [Block Parameter](../../../public-networks/how-to/use-besu-api/json-rpc.md#block-parameter)
```
Comment (alexandratran) [inline docs/private-networks/reference/api/index.md:1291]:
```suggestion
- `blockNumber`: _string_ - hexadecimal or decimal integer representing a block number or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [block parameter](../../../public-networks/how-to/use-besu-api/json-rpc.md#block-parameter)
```

File: docs/private-networks/reference/api/index.md [hunk 7]
```diff
@@ -1960,9 +1960,9 @@ Provides the following validator metrics for the specified range:
 
 #### Parameters
 
-- `fromBlockNumber`: _string_ - integer representing a block number or the string tag `earliest` as described in [Block Parameter](../../../public-networks/how-to/use-besu-api/json-rpc.md#block-parameter)
+- `fromBlockNumber`: _string_ - hexadecimal integer representing a block number or the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [Block Parameter](../../../public-networks/how-to/use-besu-api/json-rpc.md#block-parameter)
```
Comment (alexandratran) [inline docs/private-networks/reference/api/index.md:1963]:
```suggestion
- `fromBlockNumber`: _string_ - hexadecimal or decimal integer representing a block number or the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [block parameter](../../../public-networks/how-to/use-besu-api/json-rpc.md#block-parameter)
```

File: docs/private-networks/reference/api/index.md [hunk 8]
```diff
@@ -1960,9 +1960,9 @@ Provides the following validator metrics for the specified range:
...
 
-- `toBlockNumber`: _string_ - integer representing a block number or one of the string tags `latest` or `pending`, as described in [Block Parameter](../../../public-networks/how-to/use-besu-api/json-rpc.md#block-parameter)
+- `toBlockNumber`: _string_ - hexadecimal integer representing a block number or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [Block Parameter](../../../public-networks/how-to/use-besu-api/json-rpc.md#block-parameter)
```
Comment (alexandratran) [inline docs/private-networks/reference/api/index.md:1965]:
```suggestion
- `toBlockNumber`: _string_ - hexadecimal or decimal integer representing a block number or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [block parameter](../../../public-networks/how-to/use-besu-api/json-rpc.md#block-parameter)
```

File: docs/private-networks/reference/api/index.md [hunk 9]
```diff
@@ -2070,7 +2070,7 @@ Lists the validators defined in the specified block.
 
 #### Parameters
 
-- `blockNumber`: _string_ - integer representing a block number or one of the string tags `latest`, `earliest`, or `pending`, as described in [Block Parameter](../../../public-networks/how-to/use-besu-api/json-rpc.md#block-parameter)
+- `blockNumber`: _string_ - hexadecimal or decimal integer representing a block number or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [Block Parameter](../../../public-networks/how-to/use-besu-api/json-rpc.md#block-parameter)
```
Comment (alexandratran) [inline docs/private-networks/reference/api/index.md:2073]:
```suggestion
- `blockNumber`: _string_ - hexadecimal or decimal integer representing a block number or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [block parameter](../../../public-networks/how-to/use-besu-api/json-rpc.md#block-parameter)
```

File: docs/public-networks/reference/api/index.md [hunk 10]
```diff
@@ -231,9 +231,9 @@ Removes cache files for the specified range of blocks.
 
 #### Parameters
 
-- `fromBlock`: _string_ - integer representing a block number or one of the string tags `latest`, `earliest`, or `pending`, as described in [block parameter](../../how-to/use-besu-api/json-rpc.md#block-parameter)
+- `fromBlock`: _string_ - hexadecimal integer representing a block number or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [block parameter](../../how-to/use-besu-api/json-rpc.md#block-parameter)
```
Comment (alexandratran) [inline docs/public-networks/reference/api/index.md:234]:
```suggestion
- `fromBlock`: _string_ - hexadecimal or decimal integer representing a block number or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [block parameter](../../how-to/use-besu-api/json-rpc.md#block-parameter)
```

File: docs/public-networks/reference/api/index.md [hunk 11]
```diff
@@ -231,9 +231,9 @@ Removes cache files for the specified range of blocks.
...
 
-- `toBlock`: _string_ - integer representing a block number or one of the string tags `latest`, `earliest`, or `pending`, as described in [block parameter](../../how-to/use-besu-api/json-rpc.md#block-parameter)
+- `toBlock`: _string_ - hexadecimal integer representing a block number or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [block parameter](../../how-to/use-besu-api/json-rpc.md#block-parameter)
```
Comment (alexandratran) [inline docs/public-networks/reference/api/index.md:236]:
```suggestion
- `toBlock`: _string_ - hexadecimal or decimal integer representing a block number or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [block parameter](../../how-to/use-besu-api/json-rpc.md#block-parameter)
```

File: docs/public-networks/reference/api/index.md [hunk 12]
```diff
@@ -2282,7 +2282,7 @@ Returns base fee per gas and transaction effective priority fee per gas history
 
 - `blockCount`: _integer_ or _string_ - Number of blocks in the requested range. Between 1 and 1024 blocks can be requested in a single query. If blocks in the specified block range are not available, then only the fee history for available blocks is returned. Accepts hexadecimal or integer values.
 
-- `newestBlock`: _string_ - Integer representing the highest number block of the requested range or one of the string tags `latest`, `earliest`, or `pending`, as described in [block parameter](../../how-to/use-besu-api/json-rpc.md#block-parameter).
+- `newestBlock`: _string_ - Hexadecimal integer representing the highest number block of the requested range or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [block parameter](../../how-to/use-besu-api/json-rpc.md#block-parameter).
```
Comment (alexandratran) [inline docs/public-networks/reference/api/index.md:2285]:
```suggestion
- `newestBlock`: _string_ - hexadecimal or decimal integer representing the highest number block of the requested range or one of the string tags `latest`, `earliest`, `pending`, `finalized`, or `safe`, as described in [block parameter](../../how-to/use-besu-api/json-rpc.md#block-parameter).
```

File: docs/public-networks/reference/api/index.md [hunk 13]
```diff
@@ -4534,7 +4534,7 @@ None
 
 - `target`: _string_ - 32-byte required target boundary condition: 2^256 / difficulty
 
-- `blockNumber`: _string_ - hexadecimal integer representing the current block number
+- `blockNumber`: _string_ - hexadecimal or decimal integer representing the current block number
```
Comment (alexandratran) [inline docs/public-networks/reference/api/index.md:4537]:
```suggestion
- `blockNumber`: _string_ - hexadecimal integer representing the current block number
```

I assume the return values are specifically defined so this description doesn't need an update.

**Review by alexandratran — COMMENTED** [review]
Added some suggestions for missed edits. You can go to **Files changed** on this PR, and for each suggestion, **Add suggestion to batch**. Then simply commit all my suggestions at once.

**Pull request merged**